From bf1098b8a563b754e920566466c43c31f30f5afc Mon Sep 17 00:00:00 2001
From: Alex Suykov <alex.suykov@gmail.com>
Date: Wed, 19 Oct 2016 23:06:34 +0300
Subject: [PATCH] avoid redifinition of uint_least32_t on rv32

No idea why that conditional code is there in the first place.
No idea whether the patch breaks anything.
---
 libstdc++-v3/include/std/type_traits | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/libstdc++-v3/include/std/type_traits b/libstdc++-v3/include/std/type_traits
index 3a2b546..f505185 100644
--- a/libstdc++-v3/include/std/type_traits
+++ b/libstdc++-v3/include/std/type_traits
@@ -38,15 +38,7 @@
 #include <bits/c++config.h>
 
 #ifdef _GLIBCXX_USE_C99_STDINT_TR1
-# if defined (__UINT_LEAST16_TYPE__) && defined(__UINT_LEAST32_TYPE__)
-namespace std
-{
-  typedef __UINT_LEAST16_TYPE__ uint_least16_t;
-  typedef __UINT_LEAST32_TYPE__ uint_least32_t;
-}
-# else
 #  include <cstdint>
-# endif
 #endif
 
 namespace std _GLIBCXX_VISIBILITY(default)
-- 
2.10.0

